Source/Core/DolphinQt2/GameList/GameFileCache.cpp

```diff
@@ -0,0 +1,92 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "GameFileCache.h"
+
+#include <QDataStream>
+#include <QDir>
+#include <QFileInfo>
+
+#include "Common/FileUtil.h"
+#include "Core/ConfigManager.h"
+#include "DolphinQt2/Settings.h"
+
+static const int CACHE_VERSION = 1;  // Last changed in PR #5927
+static const int DATASTREAM_VERSION = QDataStream::Qt_5_0;
+
+GameFileCache::GameFileCache()
+    : m_path(QString::fromStdString(File::GetUserPath(D_CACHE_IDX) + "qt_gamefile.cache"))
+{
+}
+
+bool GameFileCache::IsCached(const QString& path)
+{
+  std::lock_guard<std::mutex> guard(m_mutex);
+
+  return m_gamefiles.contains(path);
+}
+
+GameFile GameFileCache::GetFile(const QString& path)
+{
+  std::lock_guard<std::mutex> guard(m_mutex);
+
+  return m_gamefiles[path];
+}
+
+void GameFileCache::Load()
+{
+  std::lock_guard<std::mutex> guard(m_mutex);
+
+  QFile file(m_path);
+
+  if (!file.open(QIODevice::ReadWrite))
+    return;
+
+  QDataStream stream(&file);
+  stream.setVersion(DATASTREAM_VERSION);
+
+  qint32 cache_version;
+  stream >> cache_version;
+
+  // If the cache file is using an older version, delete it and create it from scratch
+  if (cache_version != CACHE_VERSION)
+  {
+    file.remove();
```

Comment:
There's no need to remove the file, it'll be truncated and overwritten later in `Save()`.